introspection-engine/introspection-engine-tests/tests/re_introspection/relation_mode/mssql.rs

```diff
@@ -0,0 +1,361 @@
+use indoc::indoc;
+use introspection_engine_tests::test_api::*;
+
+// referentialIntegrity = "prisma" preserves the relation policy ("prisma") as well as @relations.
+#[test_connector(tags(Mssql))]
+async fn referential_integrity_prisma(api: &TestApi) -> TestResult {
+    let init = formatdoc! {r#"
+        CREATE TABLE [dbo].[Foo] (
+            [id] INT NOT NULL,
+            [bar_id] INT NOT NULL,
+            CONSTRAINT [Foo_pkey] PRIMARY KEY CLUSTERED ([id]),
+            CONSTRAINT [Foo_bar_id_key] UNIQUE NONCLUSTERED ([bar_id])
+        );
+        
+        CREATE TABLE [dbo].[Bar] (
+            [id] INT NOT NULL,
+            CONSTRAINT [Bar_pkey] PRIMARY KEY CLUSTERED ([id])
+        );
+    "#};
+
+    api.raw_cmd(&init).await;
+
+    let input = indoc! {r#"
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+            provider             = "sqlserver"
+            url                  = env("TEST_DATABASE_URL")
+            referentialIntegrity = "prisma"
+        }
+
+        model Foo {
+            id     Int @id
+            bar    Bar @relation(fields: [bar_id], references: [id])
+            bar_id Int @unique
+        }
+
+        model Bar {
+            id  Int  @id
+            foo Foo?
+        }
+    "#};
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+          provider             = "sqlserver"
+          url                  = env("TEST_DATABASE_URL")
+          referentialIntegrity = "prisma"
+        }
+
+        model Foo {
+          id     Int @id
+          bar    Bar @relation(fields: [bar_id], references: [id])
+          bar_id Int @unique
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
```

Comment:
So this is the same as the input but with some whitespace difference?
I'm curious is we can "improve" these tests:
Could we have this done as input must equal output to make it clearer / minimize the number of lines?

Reply:
In some cases this would not valid, as the line with a `@relation` is sometimes moved towards the bottom of the model.
I would defer this kind of "whitespace" optimizations to when we're sure re-introspection is an idempotent operation.